### ISSUE TITLE
NPE within NullAway using 'CheckOptionalEmptiness=true'

### ISSUE THREAD

**Issue by carterkozak** [uber/NullAway#676]
This bug occurs when using `CheckOptionalEmptiness` in combination with `Optional.ofNullable(map.get(KEY))` followed in some way by a method reference (e.g. `Map.Entry::getKey`). The failure is due to the `OptionalEmptinessHandler` `OPTIONAL_CONTENT` VariableElement returning null for all methods unexpectedly, including `VariableElement::getKind`. There are a few ways we could prevent future regressions, but that discussion may work better in a PR than this ticket.

I have a test which reproduces the failure, which I will add in a PR once I have an issue number from this report.

Environment: error-prone 2.15.0, nullaway 2.10.2. Reproducible using Java 11 and 17

```
An unhandled exception was thrown by the Error Prone static analysis plugin.
     Please report this at https://github.com/google/error-prone/issues/new and include the following:
  
     error-prone version: unknown version
     BugPattern: NullAway
     Stack Trace:
     java.lang.NullPointerException: Cannot invoke "javax.lang.model.element.ElementKind.equals(Object)" because the return value of "javax.lang.model.element.Element.getKind()" is null
  	at com.uber.nullaway.handlers.StreamNullabilityPropagator.onMatchMethodReference(StreamNullabilityPropagator.java:352)
  	at com.uber.nullaway.handlers.CompositeHandler.onMatchMethodReference(CompositeHandler.java:103)
  	at com.uber.nullaway.NullAway.matchMemberReference(NullAway.java:865)
  	at com.google.errorprone.scanner.ErrorProneScanner.processMatchers(ErrorProneScanner.java:449)
  	at com.google.errorprone.scanner.ErrorProneScanner.visitMemberReference(ErrorProneScanner.java:715)
  	at com.google.errorprone.scanner.ErrorProneScanner.visitMemberReference(ErrorProneScanner.java:150)
  	at jdk.compiler/com.sun.tools.javac.tree.JCTree$JCMemberReference.accept(JCTree.java:2546)
  	at jdk.compiler/com.sun.source.util.TreePathScanner.scan(TreePathScanner.java:86)
  	at com.google.errorprone.scanner.Scanner.scan(Scanner.java:74)
```

Thanks for using NullAway. Before you create an issue, please consider the following points:

  - [X] If you think you found a bug, please include a code sample that reproduces the problem. A test case that reproduces the issue is preferred. A stack trace alone is ok but may not contain enough context for us to address the issue.

  - [X] Please include the library version number, including the minor and patch version, in the issue text.
